pkg/deploy/generator/resources_rp.go

```diff
@@ -1311,9 +1318,8 @@ func (g *generator) rpCosmosDB() []*arm.Resource {
 		Resource:   cosmosdb,
 		APIVersion: azureclient.APIVersion("Microsoft.DocumentDB"),
 	}
-
 	if g.production {
-		cosmosdb.IPRangeFilter = to.StringPtr("[if(parameters('disableCosmosDBFirewall'), '', concat('104.42.195.92,40.76.54.131,52.176.6.30,52.169.50.45,52.187.184.26', if(equals(parameters('extraCosmosDBIPs'), ''), '', ','), parameters('extraCosmosDBIPs')))]")
```

Comment:
We are dropping these on the floor. 
We have few types of inputs here:
1. Standalone hardocoded public IPS: https://docs.microsoft.com/en-us/azure/cosmos-db/how-to-configure-firewall#allow-requests-from-the-azure-portal 
2. And private ones we provide for AME and billing.

Currently we do merge configs in our flow where if override is not provided we take one from Configuration stanza.
We need to make sure those are not gone.

They could be added into config, but we don't have "union" feature in our config merger functionality. 

With the approach you are doing now I dont from top of my head how to achieve result we need (IP from config + wellknown values) as a parameter.

Sorry for gibrish here but I think we need to step back and see how we can do this better. Maybe even changing way we construct these ARM files. :exploding_head:

Reply:
No, it makes sense. I agree that these values get lost if the value for `disableCosmosDBFIrewall` is not set and these values should _not_ be hardcoded and added to the configuration. 

The challenge here is to keep these values while allowing the other values from the template parameters to be added to the array from the configuration. 

A wider discussion may be necessary for how we do arm templates going forward.

Reply:
Yes, but we can't just copy paste save list in the config for all 30+ regions. 
This is why config merging might need to change. Currently this PR can't be merged due to this :/

Reply:
As discussed: Data is not generated but overridden in the pkg/deploy/deploy_rp.go where mgmtdocumentdb.IPAddressOrRange array is populated by the extraCosmosDBIPs from configuration as well as disableCosmosDBFIrewall parameters.